share/eds/byzantine/share_proof.go

```diff
@@ -51,6 +51,10 @@ func (s *ShareWithProof) Validate(root cid.Cid) bool {
 }
 
 func (s *ShareWithProof) ShareWithProofToProto() *pb.Share {
+	if s == nil {
+		return &pb.Share{}
+	}
```

Comment:
Is this a real case?

Reply:
It could be triggered when encoding `ErrByzantineData` that can contain nil shares now.

Reply:
I've have added that later this case is also handled on unmarshaller side